src/h5web/vis-packs/core/shared/AxisSystem.tsx

```diff
@@ -3,29 +3,37 @@ import { useThree } from '@react-three/fiber';
 import Html from './Html';
 import styles from './AxisSystem.module.css';
 import type { AxisConfig, AxisOffsets, Domain } from '../models';
-import { useFrameRendering } from '../hooks';
+import { useFrameRendering, useVisSize } from '../hooks';
 import Axis from './Axis';
 import AxisSystemContext from './AxisSystemContext';
 import { getCanvasScale } from '../utils';
 
 interface Props {
+  visRatio?: number;
   axisOffsets: AxisOffsets;
+  title?: string;
   abscissaConfig: AxisConfig;
   ordinateConfig: AxisConfig;
   children: ReactNode;
-  title?: string;
 }
 
 function AxisSystem(props: Props) {
-  const { axisOffsets, abscissaConfig, ordinateConfig, title, children } =
-    props;
+  const {
+    visRatio,
+    axisOffsets,
+    title,
+    abscissaConfig,
+    ordinateConfig,
+    children,
+  } = props;
 
   const { position, zoom } = useThree((state) => state.camera);
-  const size = useThree((state) => state.size);
-  const { width, height } = size;
+  const canvasSize = useThree((state) => state.size);
+  const { width, height } = canvasSize;
 
-  const abscissaScale = getCanvasScale(abscissaConfig, width);
-  const ordinateScale = getCanvasScale(ordinateConfig, height);
+  const visSize = useVisSize(visRatio);
+  const abscissaScale = getCanvasScale(abscissaConfig, visSize.width);
+  const ordinateScale = getCanvasScale(ordinateConfig, visSize.height);
```

Comment:
... and here. Note that in my next PR, I will inject the scales through context directly, so we don't have to recompute them in two different places (here and in `useCanvasScales`).